xarray/tests/test_dataarray.py

```diff
@@ -4079,6 +4080,49 @@ def test_rank(self):
         y = DataArray([0.75, 0.25, np.nan, 0.5, 1.0], dims=("z",))
         assert_equal(y.rank("z", pct=True), y)
 
+    # TODO: use fixture or randomly created array instead of load_dataset
+    @requires_scipy_or_netCDF4
```

Comment:
Why do we need scipy here? Because of the dataset loading?

Reply:
Yes, exactly. Until now, I was just updating the code and trying to get all tests passed. Therefore, I also added the TODO, because I think loading the dataset here is not necessary, random data or similar would be more elegant, wouldn't it?